packages/clay-core/src/tree-view/Collection.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import React from 'react';
+
+import {ItemContextProvider, useItem} from './useItem';
+
+export type ChildrenFunction<T> = (item: T) => React.ReactElement;
+
+export interface ICollectionProps<T> {
+	children: React.ReactNode | ChildrenFunction<T>;
+	items?: Array<T>;
+}
+
+function getKey(index: number, key?: React.Key | null, parentKey?: React.Key) {
+	if (
+		key != null &&
+		(!String(key).startsWith('.') || String(key).startsWith('.$'))
+	) {
+		return key;
+	}
+
+	return parentKey ? `${parentKey}.${index}` : `$.${index}`;
+}
```

Comment:
Yes, using the `key` we can even get that but for static content, I think we can skip adding reordering or DnD because it would be more complex to deal with, we would probably have to take `children` from TreeView and build a structure based on it and store in the state to allow us to manipulate and have an effect on rendering.